cpp/tests/reductions/reduction_tests.cu

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cstdlib>
+#include <iostream>
+#include <vector>
+#include <algorithm>
+#include <iterator>
+#include <type_traits>
+#include <stdlib.h>
+
+#include "gtest/gtest.h"
+
+#include <cudf.h>
+#include <cudf/functions.h>
+
+#include <thrust/device_vector.h>
+
+#include "tests/utilities/cudf_test_fixtures.h"
+#include "tests/utilities/cudf_test_utils.cuh"
+#include "tests/utilities/column_wrapper.cuh"
+
+template <typename T>
+T getValueFromScalar(gdf_scalar& scalar)
+{
+    T val;
+    memcpy(&val, &scalar.data, sizeof(T));
+
+    return val;
+}
+
+// This is the main test feature
+template <typename T>
+struct ReductionTest : public GdfTest
+{
+    // Sum/Prod/SumOfSquare never support non arithmetics
+    static constexpr bool ret_non_arithmetic =
+        (std::is_arithmetic<T>::value) ? true : false;
+
+    ReductionTest(){}
+
+    ~ReductionTest(){}
+
+    void examin(std::vector<int>& int_values,
+        T exact_value, bool succeeded_condition,
+        gdf_reduction_op op)
+    {
+        gdf_size_type col_size = int_values.size();
+        std::vector<T> input_values(col_size);
+
+        std::transform(int_values.begin(), int_values.end(),
+            input_values.begin(),
+           [](int x) { T t(x) ; return t; } );
+
+        cudf::test::column_wrapper<T> const col(input_values);
+
+        const gdf_column * underlying_column = col.get();
+        thrust::device_vector<T> dev_result(1);
+
+        auto statement = [&]() {
+            gdf_scalar result = gdf_reduction(underlying_column, op,
+                                              underlying_column->dtype);
+            T host_result = getValueFromScalar<T>(result);
+            EXPECT_EQ(exact_value, host_result);
```

Comment:
I think you can just do `EXPECT_EQ(exact_value, result.value)`, you don't need the `getValueFromScalar`.